File: app-fw/app-base/src/main/java/org/yy/base/dao/dynamic2/Expression.java
package org.yy.base.dao.dynamic2;

import java.util.ArrayList;
import java.util.List;


    public abstract class Expression
    {
        protected List<Expression> lists = new ArrayList<Expression>();
        public void Append(Expression exp)
        {
            lists.add(exp);
        }
        public abstract String Eval(ExpressionContext ctx);
    }

File: SqlParser/SqlParser/Expression.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace SqlParser
{
    public abstract class Expression
    {
        protected List<Expression> lists = new List<Expression>();
        public void Append(Expression exp)
        {
            lists.Add(exp);
        }
        public abstract string Eval(IExpressionContext ctx);
    }
}
